### DISCUSSION TITLE
Config.json Not Reading My Circle CI build

### DISCUSSION THREAD

**Discussion by asmarazaq** [test-results-reporter/reporter#205] [Q&A] [answered]
How you guys made the circle ci build work as its not working for my stuff. How can we pass the dynamic value from CircleCi Build to our config.json file.Here is my config.json file;
```json
{
  "project": "example-playwright-testbeats",
  "run": "E2E Tests",    
  "targets": [
    {
      "name": "teams",                    
      "inputs": {
        "url": "Teams Webhook",
        "title": "example-playwright-testbeats - E2E Tests",
        "only_failures": true
              
       
      },
      "extensions": [
        {
          "name": "hyperlinks",
          "inputs": {
            "links": [
              {
                "text": "Tests",
                "url": "($CIRCLE_BUILD_URL)"
              }                
            ]
          }
        },
        {
          "name": "quick-chart-test-summary"
        },
        {
          "name": "ci-info"
        }
      ]
    }
  ],    
  "results": [
    {
      "type": "junit",
      "files": ["test-results/junit.xml"]
    }
  ]
}  
```

**Comment by ASaiAnudeep** [comment, marked as answer]
Dynamic values can be passed through environment variables - `{<ENV>}`

**Docs**: https://docs.testbeats.com/others/environment-variables.html
**Example**: 
- https://github.com/test-results-reporter/example-playwright-testbeats/blob/main/testbeats.config.json#L7
- https://github.com/test-results-reporter/example-playwright-testbeats/blob/main/.github/workflows/test.yml#L27

**Comment by asmarazaq** [comment]
It reads the config.json file and adds the url if we read from config.yml in pipeline but it does not posts a circle link in Teams' channel
      "name": "teams",                    
      "inputs": {
        "url": "Teams webhook",
        "title": "example-playwright-testbeats - E2E Tests",
        "only_failures": true
              
       
      },
      "extensions": [
        {
          "name": "hyperlinks",
          "inputs": {
            "links": [
              {
                "text": "Tests",
                "url": "https://circleci.foc.zone/gh/BrokerPro/playwright"
              }                
            ]
          }
        },
        {
          "name": "quick-chart-test-summary"
        },
        {
          "name": "ci-info"
        }
      ]
    }
  ],    
  "results": [
    {
      "type": "junit",
      "files": ["test-results/junit.xml"]
    }
  ]
}

**Comment by asmarazaq** [comment]
![Screenshot 2024-07-03 123029](https://github.com/test-results-reporter/testbeats/assets/45542493/45f45a84-df3b-4149-a856-c2d8382a3420)

**Comment by asmarazaq** [comment]
I want a link of circle ci to be added in the above screenshot. Just an FYI i am not using API key

**Reply by ASaiAnudeep** [reply]
I believe if you click on **Tests**, it will redirect you to the circle ci build. If you want the url to be visible, replace `Tests` with url.
```json
{
  "name": "hyperlinks",
  "inputs": {
    "links": [
      {
        "text": "https://circleci.foc.zone/gh/BrokerPro/playwright",
        "url": "https://circleci.foc.zone/gh/BrokerPro/playwright"
      }                
    ]
  }
}
```
